core/safeclient/client.go

```diff
@@ -493,29 +356,16 @@ func (c *SafeEthClient) SubscribeNewHead(ctx context.Context, ch chan<- *types.H
 			case <-safeSub.Err():
 				c.logger.Info("Safe subscription to new heads ended")
 				return
-			case success := <-reinitC:
-				reinitC = c.WatchReinit()
-				if success {
-					err := resub()
-					if err = c.handleClientError(err); err != nil {
-						c.logger.Error("Failed to resubscribe to new heads", "err", err)
-					}
-				}
 			case <-sub.Err():
 				c.logger.Info("Underlying subscription to new heads ended, resubscribing")
-				err := resub()
-				if err = c.handleClientError(err); err != nil {
-					c.logger.Error("Failed to resubscribe to new heads", "err", err)
-				}
+				handleResub()
 			case <-headerTicker.C:
-				c.logger.Info("Header ticker fired, ending subscription")
+				c.logger.Debug("Header ticker fired")
```

Comment:
Could it be possible that the `headerTicker` + the `sub.Err` channel both emit a value and we end up running `handleResub` twice? If so, we'll end up with two subscriptions.

Reply:
That's a good point. The select is synchronous and the resub is not done in a goroutine, so that means they could only run one at a time. When the underlying subscription on `safeSub` is changed, the previous one is unsubscribed. So, in the end I don't think we're leaking anything here.

Reply:
Alright, that makes sense. I missed the part when we ensure that on `safeSub` we `Unsubscribe` so we should not be leaking anything.

Reply:
Great! One small refactor we could make is accessing the `safeSub` underlying subscription directly instead of having a local variable for it as well - makes things less confusing. Done in [386a39d](https://github.com/NethermindEth/near-sffl/pull/175/commits/386a39da426f45474b799995fee46598cf685b86)